src/casp/services/api/data_manager/cell_operations.py

```diff
@@ -111,31 +112,72 @@ def get_neighborhood_distance_summary_dev_details(
 
         return bigquery_response_parsers.parse_match_query_job(query_job=query_job, include_dev_details=True)
 
-    def get_cell_metadata_by_ids(
-        self, cell_ids: t.List[int], metadata_feature_names: t.List[str], model_name: str
-    ) -> t.List[t.Dict[str, t.Any]]:
+    def _get_cell_metadata_by_ids(
+        self, cell_ids: t.List[int], feature_names: t.List[str]
+    ) -> t.List[schemas.CellariumCellMetadata]:
         """
-        Get cells by ids.
+        Get cells by ids, maintaining the order of cell_ids. Preserves the order of cell_ids in the query results.
```

Comment:
"Get cells by ids ..." -> "Get cells by CAS IDs, maintaining the order of ``cell_ids``. Preserves the order of ``cell_ids``..."